pkg/rcache/rcache.go

```
@@ -57,190 +115,25 @@ func redisPool() (*pool.Pool, error) {
 	return connPool_, nil
 }
 
-// getConn returns a redis client from the pool. When you are done you must
-// call the cleanup function to return the connection to the pool.
-func getConn() (*redis.Client, func(), error) {
+// cmd is a helper around redis.(*Client).Cmd. As a convenience it returns
+// Resp.Err as err if we get a response. This reduces the number of error
+// checks needed.
+func cmd(cmd string, args ...interface{}) (*redis.Resp, error) {
 	connPool, err := redisPool()
 	if err != nil {
-		return nil, nil, err
+		log15.Warn("failed to connect to redis pool", "error", err)
+		return nil, err
```

It's a bit strange contract to log **and** return the error. Maybe move the log statements to where we discard the error value?